extern/detray/CMakeLists.txt

```diff
@@ -0,0 +1,41 @@
+# TRACCC library, part of the ACTS project (R&D line)
+#
+# (c) 2021 CERN for the benefit of the ACTS project
+#
+# Mozilla Public License Version 2.0
+
+# CMake include(s).
+cmake_minimum_required( VERSION 3.14 )
+include( FetchContent )
+
+# Tell the user what's happening.
+message( STATUS "Building Detray as part of the TRACCC project" )
+
+# Declare where to get Detray from.
+set( TRACCC_DETRAY_SOURCE
+   "GIT_REPOSITORY;https://github.com/acts-project/detray.git;GIT_TAG;3aafe70"
+   CACHE STRING "Source for Detray, when built as part of this project" )
+mark_as_advanced( TRACCC_DETRAY_SOURCE )
+FetchContent_Declare( Detray ${TRACCC_DETRAY_SOURCE} )
+
+# Options used in the build of Detray.
+set( DETRAY_BUILD_TESTING FALSE CACHE BOOL
+   "Turn off the build of the Detray unit tests" )
+set( DETRAY_EIGEN_PLUGIN TRUE CACHE BOOL
+   "Turn on the build of the Detray Eigen code" )
+set( DETRAY_VC_PLUGIN TRUE CACHE BOOL
+   "Turn on the build of the Detray Vc code" )
+
+set( DETRAY_SETUP_VECMEM FALSE CACHE BOOL
+   "Do not set up VecMem as part of Detray" )
+set( DETRAY_SETUP_ALGEBRA_PLUGINS FALSE CACHE BOOL
+   "Do not set up Algebra Plugins as part of Detray" )
+set( DETRAY_SETUP_DFELIBS FALSE CACHE BOOL
+   "Do not set up dfelibs as part of Detray" )
+set( DETRAY_SETUP_GOOGLETEST FALSE CACHE BOOL
+   "Do not set up GoogleTest as part of Detray" )
+set( DETRAY_SETUP_BENCHMARK FALSE CACHE BOOL
+   "Do not set up Google Benchmark as part of Detray" )
```

Comment:
This seems fragile. Not in a way we can really control on the traccc side, but perhaps we could move to a more sane automatic detection mechanism for this? Or maybe disable these by default?

Reply:
I do not agree. Tests should be enabled when we build a project as a top-level project.

Many projects that I've seen test whether they are the top level projects, and then disable their tests by default if they are not. We could potentially go for something like that with these projects.

But I myself don't see a huge issue with how these flags are set at the moment. The same type of complexity was there before my changes as well. The traccc build system was just oblivious to them. (And was not able to affect the behaviour of the upstream projects, as it can with the latest versions of those.)

Reply:
I think checking whether something is a top-level project is also an acceptable solution. The problem I see with defaulting all these things on be enabled is that you will need to update the traccc configuration whenever one of the dependencies adds any additional flags. Sounds like technical debt that we might want to avoid.